aten/src/ATen/functorch/BatchRulesPooling.cpp

```diff
@@ -48,7 +78,9 @@ TORCH_LIBRARY_IMPL(aten, FuncTorchBatched, m) {
   EXISTING_BDIM_ALL_BOXED(adaptive_max_pool3d);
   ALL_TENSORS_HAVE_OPTIONAL_BDIM_BOXED_CONTIG1(3, adaptive_max_pool2d_backward, 2);
   ALL_TENSORS_HAVE_OPTIONAL_BDIM_BOXED_CONTIG1(4, adaptive_max_pool3d_backward, 2);
-
+  // We can get max_pool1d to work on CUDA through decomposition,
+  // but fails on CPU due to max_pool1d_cpu not having a derivative.
```

Comment:
Hi @zou3519, there is a reason not using decomposition here, where the comment was originally written in test_vmap.py.

Reply:
The comment in test_vmap.py is outdated :). 

When a functorch transform is active, max_pool1d_impl (the problematic piece of the implementation) never gets invoked:
https://github.com/pytorch/pytorch/blob/db456ab83da6a505dcebc128903d5ee4fc2d5712/aten/src/ATen/native/MaxPooling.cpp#L146-L155

Concretely this happens because under vmap, the input tensors are wrapped in a BatchedTensor Tensor subclass, and there is a check for `isTensorSubclassLike` in the code.

Reply:
ok, let me try to decompose it :)

Reply:
Done.